### ISSUE TITLE
Missing bounds check

### ISSUE THREAD

**Issue by kaboomium** [hunspell/hunspell#452]
We're seeing that, by throwing the word, "X4444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444444XXXX" against suggestions (with en-us, if that matters), that we get an array overrun. We have found that if we make the enclosed change, that we avoid the error. I didn't make a pull request because maybe you want to incorporate MAXWORDLEN, or reduce the 200.

```diff
--- a/src/hunspell/suggestmgr.cxx
+++ b/src/hunspell/suggestmgr.cxx
@@ -1493,10 +1493,10 @@ int SuggestMgr::checkword(const std::string& word,
     int nosuffix = 0;
 
     if (cpdsuggest == 1) {
-      if (pAMgr->get_compound()) {
+      if (pAMgr->get_compound() && word.size() < 100) {
         struct hentry* rv2 = NULL;
-        struct hentry* rwords[100];  // buffer for COMPOUND pattern checking
-        rv = pAMgr->compound_check(word, 0, 0, 100, 0, NULL, (hentry**)&rwords, 0, 1, 0);  // EXT
+        struct hentry* rwords[200];  // buffer for COMPOUND pattern checking
+        rv = pAMgr->compound_check(word, 0, 0, 200, 0, NULL, (hentry**)&rwords, 0, 1, 0);  // EXT
         if (rv &&
             (!(rv2 = pAMgr->lookup(word.c_str())) || !rv2->astr ||
              !(TESTAFF(rv2->astr, pAMgr->get_forbiddenword(), rv2->alen) ||
```

**Comment by dimztimz** [comment]
Probably duplicate of #102. I already fixed it.

**Comment by kaboomium** [comment]
I'm sync'd to 65883dd5 (a week ago.)

**Comment by dimztimz** [comment]
I made an error in my previews comment which I fixed. I meant duplicate of issue 446.

Additional I retested with Address Sanitizer your case before the fix and after the fix in #446. Before the fix I get error, after I don't, as expected.

I would advise you to retest with the recently released 1.6.0 and close this if it's resolved there. I will mark it as duplicate for now.

**Comment by kaboomium** [comment]
That change does fix it.
